guby_backend/views.py

```diff
@@ -66,10 +76,17 @@ def meetup_delete(request, meetup_id):
 
     return render(request, 'app/meetup_delete.html', {"form": form, "meetup_id": meetup.id})
 
+@login_required(login_url='/app/users/login/')
 def event_index(request, meetup_id):
-    events = Event.objects.filter(meetup=meetup_id)
-    return render(request, 'app/events.html', {'meetup_id': meetup_id, 'events': events})
-
+    # check if login user is owner of meetupid
+    meetup = Meetup.objects.get(id=meetup_id)
+    if meetup.creator.id == request.user.id:
+        events = Event.objects.filter(meetup=meetup_id)
+        return render(request, 'app/events.html', {'meetup_id': meetup_id, 'events': events})
+    else:
+        raise Http404("Event does not exist") 
```

Comment:
I would've raised 403. But 404 makes more sense here. If they're not the creator, then why should they even know of this event existing ✨ 👏